frontend/ui/timeval.lua

```diff
@@ -1,10 +1,43 @@
+--[[--
+A simple module to deal with time values.
+
+@usage
+    local TimeVal = require("ui/timeval")
+
+    local tv_start = TimeVal:now()
+    -- Do some stuff.
+    -- You can add and substract `TimeValue` objects.
+    local tv_duration = TimeVal:now() - tv_start
+    -- If you need more precision (like 2.5 s),
+    -- you can add the milliseconds to the seconds.
+    local tv_duration = tv_duration.sec + tv_duration.usec/1000000
```

Comment:
A bit confusing: your first tv_duration is still a TimeVal object (which can be used for other TimeVal arithmetic).
The second is a number (whether you add or nor the `.usec` if you need more precision).
(idem below when documenting TimeVal:now())

Reply:
The confusing example comes straight from your code, lol :-P

Reply:
:) ok, but if i'm dirty changing a variable type just to not introduce an additional local variable (i often do that), that should not be documented :)